src/main/scala/geotrellis/test/validation/MultibandSpaceTime.scala

```diff
@@ -0,0 +1,143 @@
+    val controlRaster = controlTiff.raster
+    val controlSampleExtent = randomExtentWithin(controlRaster.extent, jConfig.validationOptions.sampleScale)
+
+    // Transformations
+    val transformation = Transform(controlTiff.crs, jConfig.ingestOptions.layoutScheme.crs)
+    val invTransformation = Transform(jConfig.ingestOptions.layoutScheme.crs, controlTiff.crs)
+
+    // Test parameters
+    val diffThreshold = jConfig.validationOptions.resolutionThreshold
+    val ingestSampleExtent = controlSampleExtent.reproject(transformation)
+
+    // Test values
+    val testRaster = read(layerId, None)
+      .filter().where(Intersects(ingestSampleExtent)).result
+      .stitch(dt)
+
+
+    // Here, we need to:
+    // 1. get the RasterExtent from our testRaster
+    // 2. iterate over its cols/rows
+    //   a. get the value from our testRaster at that location
+    //   b. get the value from our controlRaster at the projection of that location
+    // 3. Compare those values
+    def controlIndicesFromTest(col: Int, row: Int): (Int, Int) = {
+      val (x, y) = testRaster.rasterExtent.gridToMap(col, row)
+      val (xprime, yprime) = invTransformation(x, y)
+      controlRaster.rasterExtent.mapToGrid(xprime, yprime)
+    }
+
+    for (band <- 0 until testRaster.bandCount) {
+
+      // Counts (for later)
+      val cellCount = testRaster.rasterExtent.cols * testRaster.rasterExtent.rows // cell count
+      var diffTotal = 0.0 // Sum of test vs control values
+      var nanCount = 0 // Count of NaN values encountered
+      var outOfBoundsCount = 0
+      // Useful stats (for later)
+      var minControl = 0.0
+      var maxControl = 0.0
+      var minTest = 0.0
+      var maxTest = 0.0
+
+      for (testCol <- 0 until testRaster.rasterExtent.cols) {
+        for (testRow <- 0 until testRaster.rasterExtent.rows) {
+          try {
+            val (controlCol, controlRow) = controlIndicesFromTest(testCol, testRow)
+            val controlValue = controlRaster.band(band).getDouble(controlCol, controlRow)
+            val testValue = testRaster.band(band).getDouble(testCol, testRow)
+
+            if (isData(controlValue) && isNoData(testValue)) nanCount += 1
+            else if (isNoData(controlValue) && isData(testValue)) nanCount += 1
+            else if (isNoData(controlValue) && isNoData(testValue)) ()
+            else {
+              minControl = min(minControl, controlValue)
+              maxControl = max(maxControl, controlValue)
+              minTest = min(minTest, testValue)
+              maxTest = max(maxTest, testValue)
+              diffTotal += abs(controlValue - testValue)
+            }
+          } catch {
+            case e: java.lang.ArrayIndexOutOfBoundsException =>
+              outOfBoundsCount += 1
+          }
+        }
+      }
+      logger.info(s"Resample correctness band ${band + 1}")
+      if (outOfBoundsCount > 0) logger.warn(s"Index out of bounds errors encounted: $outOfBoundsCount exceptions")
```

Comment:
This is noteworthy - I'm not sure why this sometimes throws an exception. Perhaps we're dealing with some floating point precision issues. This happens rarely and randomly.